### ISSUE TITLE
[Bug] Problem using VecNormalize and set_env() function when using SAC and TD3

### ISSUE THREAD

**Issue by cleversonahum** [DLR-RM/stable-baselines3#637]
### 🐛 Bug

Using set_env() function with SAC and TD3 generates the error `could not broadcast input array from shape (0) into shape (X,Y)` when the environment is wrapped with VecNormalize. The set_env() works correctly using PPO and A2C, so maybe it is a specific problem with off-policy algorithms. 

### To Reproduce

I've created a custom environment where I first start the agent with a "dummy" environment and after the correct environment is set using set_env() inside a loop. This structure has worked for PPO and A2C algorithms, but not for SAC and TD3. A simple environment and training that triggers this error is presented below:

```python
from numpy.core.numeric import Inf
from stable_baselines3.common.vec_env import DummyVecEnv, VecNormalize
from stable_baselines3 import SAC
from gym import spaces
import gym
import numpy as np


class Test(gym.Env):
    metadata = {"render.modes": ["human"]}

    def __init__(self):
        # super(Test, self).__init__()
        self.action_space = spaces.Box(low=-1, high=1, shape=(3,), dtype=np.float32)
        self.observation_space = spaces.Box(
            low=0, high=np.inf, shape=(125,), dtype=np.float32
        )

    def reset(self):
        return self.observation_space.sample()

    def step(self, action):
        reward = 0.0
        done = False
        return self.observation_space.sample(), reward, done, {}


env = Test()
env1 = Test()
env = DummyVecEnv([lambda: env])
env = VecNormalize(env)
model = SAC("MlpPolicy", env, verbose=1)
env1 = DummyVecEnv([lambda: env1])
env1 = VecNormalize(env1)
model.set_env(env1)
model.learn(1900)
```

I received the following error:
```
Exception has occurred: ValueError       (note: full exception trace is shown but execution is paused at: <module>)
could not broadcast input array from shape (0) into shape (1,125)
  File "/home/cleversonahum/Documents/rrm-rl/rrm-slice-rl/stable-baselines3/stable_baselines3/common/buffers.py", line 232, in add
    self.observations[self.pos] = np.array(obs).copy()
  File "/home/cleversonahum/Documents/rrm-rl/rrm-slice-rl/stable-baselines3/stable_baselines3/common/off_policy_algorithm.py", line 506, in _store_transition
    infos,
  File "/home/cleversonahum/Documents/rrm-rl/rrm-slice-rl/stable-baselines3/stable_baselines3/common/off_policy_algorithm.py", line 591, in collect_rollouts
    self._store_transition(replay_buffer, buffer_action, new_obs, reward, done, infos)
  File "/home/cleversonahum/Documents/rrm-rl/rrm-slice-rl/stable-baselines3/stable_baselines3/common/off_policy_algorithm.py", line 361, in learn
    log_interval=log_interval,
  File "/home/cleversonahum/Documents/rrm-rl/rrm-slice-rl/stable-baselines3/stable_baselines3/sac/sac.py", line 300, in learn
    reset_num_timesteps=reset_num_timesteps,
  File "/home/cleversonahum/Documents/rrm-rl/rrm-slice-rl/test_env.py", line 36, in <module> (Current frame)
    model.learn(1900)
```

The code works both If I change from SAC to A2C or PPO, or if I comment on the DummyVecEnv and VecNormalize lines.

### Expected behavior

I expected the same behavior from A2C and PPO execution but it seems that the variable `obs` is empty because of something related to the wrapping process done into set_env().

### System Info

-  OS: Linux-5.10.70-1-MANJARO-x86_64-with-arch-Manjaro-Linux 1 SMP PREEMPT Thu Sep 30 15:29:01 UTC 2021
-  Python: 3.7.12
-  Stable-Baselines3: 1.3.0
-  PyTorch: 1.9.0+cu102
-  GPU Enabled: True
-  Numpy: 1.19.5
-  Gym: 0.18.3

### Additional context
The reason I am using set_env() in the training process is that I wanted to vary the custom environment characteristics to provide RL training on different scenarios.

**Comment by araffin** [comment]
Hello,
in you code example why are you changing the env even before training? in that case, why don't you start with `env1`?

if you train and then change the env, it does work.

```python
model = SAC("MlpPolicy", env, verbose=1, policy_kwargs=dict(net_arch=[32]))
model.learn(200)
model.set_env(env1)
model.learn(200)

```

**Comment by araffin** [comment]
I see where the bug comes from... will push a fix soon

**Comment by araffin** [comment]
Should be fixed in https://github.com/DLR-RM/stable-baselines3/pull/638

**Comment by cleversonahum** [comment]
Hello @araffin, the code I've appended is just a simple example to cause the error. In the main code, I am setting the environment into a loop that varies between environment conditions and RL methods. So I first set a dummy environment (just to initialize the model), and into the loop, I change to the correct environment. It was just a way that I find to automatize the simulation. Thx for the help. I will follow the PR #638 .
